### PR TITLE
add Session#screenshot, makefile install deps, some test

### PR DESCRIPTION
**Pull request by codenoid**

### CODE REVIEW

**Comment by codenoid** [comment]
lol, sorry for my crystal auto indent

File: src/webdriver/session.cr [hunk 1]
```diff
@@ -77,40 +77,42 @@ module Selenium
...
 
     def parent_frame
       post("/frame/parent")
     end
 
-    #def screenshot
-    #  # TODO: decode Base64
-    #  post("/screenshot")
-    #end
+    def screenshot(path = "./screenshot.png")
```
Comment (codenoid) [inline src/webdriver/session.cr]: or better with

```
      if File.writable?(path)
        File.write(path, decoded_image)
      else
        pp "Cannot write file to current directory"
        return false
      end
```

@ysbaddaden

Comment (ysbaddaden) [reply src/webdriver/session.cr]: I'd say to let it raise, a developer should be capable to deal with the exception.

**Comment by ysbaddaden** [comment]
Yes, can you avoid the unrelated format changes? The patch would be simpler :-)

File: src/webdriver/session.cr (hunk 1 above)
Comment (codenoid) [reply src/webdriver/session.cr]: you mean, let `File.write` raise his error & exception ? okay

**Comment by codenoid** [comment]
okay, next @ysbaddaden ?

File: test/session_test.cr [hunk 2]
```diff
@@ -40,7 +44,9 @@ class SessionTest < Minitest::Test
     count = types.size
 
     input = session.find_element(:css, "input[type='search']")
-    input.send_keys("Foo bar.., ups wrong...")
+    input.send_keys("Cleint.., ups wrong...")
+    sleep 1
```
Comment (ysbaddaden) [inline test/session_test.cr]: Any reason to introduce a sleep, here?

**Comment by ysbaddaden** [comment]
Looks good!

I'm just wondering whether we should always save the image, or just return the decoded slice. Maybe have both with `#screenshot` and `#save_screenshot(path)`?

By the way, we can optimize things a bit, for example decode directly to a file:

```crystal
def screenshot
  data = get("/screenshot").as(String)
  Base64.decode(data)
end

def save_screenshot(path)
  data = get("/screenshot").as(String)
  File.open(path, "w") { |file| Base64.decode(data, file) }
end
```

File: test/session_test.cr (hunk 2 above)
Comment (codenoid) [reply test/session_test.cr]: umm, users can view posts before (and delay ...) -> .clear -> after

**Comment by codenoid** [comment]
yeah, more option, thanks

**Comment by ysbaddaden** [comment]
Thank you!

**Pull request merged**